modules/menus/calendar/index.ts

```diff
@@ -19,7 +22,13 @@ export default () => {
             Widget.Box({
               class_name: "calendar-content-items",
               vertical: true,
-              children: [TimeWidget(), CalendarWidget(), WeatherWidget()],
+              children: weatherEnabled.bind("value").as(isWeatherEnabled => {
```

Comment:
While this works, I've discovered that you want to decouple binding as much as possible. In this case, the Time and Calendar are both coupled with that binding.

What we can do instead is move the binding into the WeatherWidget and render an empty box if the weather is disabled. That way the binding only effects the component it's meant to.

Everything else looks good though, you've got the right idea :+1:

Reply:
Personally, I wouldn't expect something like WeatherWidget to render an empty Box, because the clock menu is configured to not include it. 

My idea of the option was that it should be used to exclude it from the clock menu and not disable weather at all. The weather widget is independent and could also be used in a dedicated weather bar module.

Reply:
If the empty box is the wanted behavior, then the setting should be called differently

Reply:
By empty box I mean returning Widget.Box() which renders nothing at all. Visually it's the same effect as what you have, but if we want enable granularity in picking specific modules and reusing them then I think this approach is fine.